File: NewsApp/app/src/main/java/com/example/newsapp/Activity/MainActivity.java
package com.example.newsapp.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.util.Log;

import com.android.volley.AuthFailureError;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import com.example.newsapp.adaptor.ViewPagerAdapter;
import com.example.newsapp.fragment.IndiaFragment;
import com.example.newsapp.fragment.TechFragment;
import com.example.newsapp.fragment.USAFragment;
import com.example.newsapp.model.NewsData;
import com.example.newsapp.adaptor.RecyclerAdapter;
import com.example.newsapp.databinding.ActivityMainBinding;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;
    RecyclerAdapter adapter;
    ArrayList<NewsData> newsDataList;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ViewPagerAdapter vpAdapter = new ViewPagerAdapter(getSupportFragmentManager());

        vpAdapter.addFragment(new IndiaFragment(), "India");
        vpAdapter.addFragment(new USAFragment(), "USA");
        vpAdapter.addFragment(new TechFragment(),"Indonesia");

        binding.viewPager.setAdapter(vpAdapter);
        binding.tabLayout.setupWithViewPager(binding.viewPager);




    }
}

File: RecyclerView App/RecyclerViewExemplar/app/src/main/java/com/akshatbhuhagal/recyclerviewexemplar/MainActivity.kt
package com.akshatbhuhagal.recyclerviewexemplar

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import com.akshatbhuhagal.recyclerviewexemplar.databinding.ActivityMainBinding
import java.util.ArrayList

class MainActivity : AppCompatActivity(), MyAdapter.ItemClicked {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.recyclerView.layoutManager = LinearLayoutManager(this)

        val items = fetchData()
        val adapter = MyAdapter(items, this)

        binding.recyclerView.adapter = adapter

    }

    private fun fetchData() : ArrayList<String> {
        val list = ArrayList<String>()
        for (i in 0 until 100) {
            list.add("Item $i")
        }
        return list
    }

    override fun onItemClicked(item: String) {
        Toast.makeText(this, "Clicked on $item", Toast.LENGTH_SHORT).show()
    }

}